packages/wrangler/src/containers/containers.ts

```diff
@@ -38,33 +38,51 @@ export async function deleteCommand(
 			"You must provide an ID. Use 'wrangler containers list` to view your containers."
 		);
 	}
-	if (deleteArgs.json || !isInteractive()) {
+
+	if (deleteArgs.json) {
 		const container = await ApplicationsService.deleteApplication(
 			deleteArgs.ID
 		);
 		console.log(JSON.stringify(container, null, 4));
 		return;
 	}
+
 	startSection("Delete your container");
-	const yes = await inputPrompt({
-		question:
-			"Are you sure that you want to delete these containers? The associated DO container will lose access to the containers.",
-		type: "confirm",
-		label: "",
-	});
-	if (!yes) {
-		cancel("The operation has been cancelled");
-		return;
+	if (isInteractive()) {
+		const yes = await inputPrompt({
+			question:
+				"Are you sure that you want to delete these containers? The associated DO container will lose access to the containers.",
+			type: "confirm",
+			label: "",
+		});
+		if (!yes) {
+			cancel("The operation has been cancelled");
+			return;
+		}
 	}
 
 	const [, err] = await wrap(
 		ApplicationsService.deleteApplication(deleteArgs.ID)
 	);
 	if (err) {
+		if (err instanceof ApiError) {
+			if (err.status === 400 || err.status === 404) {
+				const body = JSON.parse(err.body);
+				throw new UserError(
+					`There has been an error deleting the container.\n${body.error}`
+				);
+			}
+
+			throw new UserError(
+				`There has been an unknown error deleting the container.\n${JSON.stringify(err.body)}`
+			);
```

Comment:
This doesn't seems like a user error, how about making it a `FatalError`?

Reply:
FatalError is also a UserError, just with an additional exitCode.

Reply:
In this case I would just rethrow the APIError?

Reply:
Or just an `Error`